pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,95 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	otbridge "go.opentelemetry.io/otel/bridge/opentracing"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT     opentracing.Tracer
+	OTEL   trace.TracerProvider
+	closer func() error
 }
 
-func OT(t opentracing.Tracer) JTracer {
-	return JTracer{OT: t}
+var once sync.Once
+
+func New() (*JTracer, error) {
+	jt := JTracer{}
+	opentracingTracer, otelTracerProvider, closed, error := jt.initBoth(context.Background())
```

Comment:
There are a lot of returns from this method, and also it doesn't need to have a receiver. 

I have a suggestion for breaking this up a little to make it a little easier to read:

1. create a func (e.g. `initOTEL`) that returns just the otel tracer provider. Remove `initBoth`.
2. build the opentracing tracer separately from within `New`
3. build the closer separately from within `New`

Also, the `otlp` serviceName we saw in the failing integration test was due to this config below:
```go
semconv.ServiceNameKey.String("otlp")
```

It seems the resource attributes (that contain the service name) need to be determined when the tracer provider is created, so we may need to pass in the service name into `New`. This implies a tracer provider per service.

Try to avoid using `error` as a variable name; it's the type name for the error interface, so it can be pretty confusing if overloaded as a var name.

We don't need `jt` and `initOTEL` does not need to have the `JTracer` as a receiver.

So putting all that together:
```go
func New(serviceName string) (*JTracer, error) {
	ctx := context.Background()
	tracerProvider, err := initOTEL(ctx, serviceName)
	if err != nil {
		return nil, err
	}
	// Use the bridgeTracer as your OpenTracing tracer(otTrace).
	otelTracer := tracerProvider.Tracer("github.com/jaegertracing/jaeger/pkg/jtracer")
	otTracer, wrappedOTELTracer := otbridge.NewTracerPair(otelTracer)

	closer := func() error {
		return tracerProvider.Shutdown(ctx)
	}

	return &JTracer{
		OT:     otTracer,
		OTEL:   wrappedOTELTracer,
		closer: closer,
	}, nil
}
```

Reply:
small nit: wrappedOTELTracer -> wrappedTracerProvider

Reply:
another nit: the closer function should take context as argument provided by the caller